sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -447,6 +447,12 @@ object GpuDeviceManager extends Logging {
     }
     // Host memory limits must be set after the pinned memory pool is initialized
     HostAlloc.initialize(nonPinnedLimit)
+    // Fill the MULTITHREAD_READ_MEM_LIMIT with the 90% of the total OFF_HEAP memory
+    // if it is not set already.
+    if (conf.multiThreadMemoryLimit == 0) {
+      sparkConf.set(RapidsConf.MULTITHREAD_READ_MEM_LIMIT.key,
```

Comment:
Is this code actually guaranteed to execute before this conf is used, for instance in the GpuReaderFactory?

Reply:
I think so. This piece of code will be executed during the initialization of Executor Plugin, ahead of the query planning.